app/models/analysis_jobs_item.rb

```diff
@@ -0,0 +1,197 @@
+
+
+  # association validations
+  validates :analysis_job, existence: true
+  validates :audio_recording, existence: true
+
+  # attribute validations
+  validates :status, presence: true, length: {minimum: 2, maximum: 255}
+  validates :queue_id, uniqueness: {case_sensitive: true}
+
+  validate :is_new_when_created, on: :create
+  validate :has_queue_id_when_needed
+
+  validates :created_at,
+            presence: true,
+            timeliness: {on_or_before: lambda { Time.zone.now }, type: :datetime},
+            unless: :new_record?
+
+  validates :queued_at, :work_started_at, :completed_at,
+            allow_blank: true, allow_nil: true,
+            timeliness: {on_or_before: lambda { Time.zone.now }, type: :datetime}
+
+  # item status values - timed out is a special failure case where the worker never reports back
+  AVAILABLE_ITEM_STATUS_SYMBOLS = [:new, :queued, :working, :successful, :failed, :timed_out, :cancelled]
+  AVAILABLE_ITEM_STATUS = AVAILABLE_ITEM_STATUS_SYMBOLS.map { |item| item.to_s }
+
+  AVAILABLE_JOB_STATUS_DISPLAY = [
+      {id: :new, name: 'New'},
+      {id: :queued, name: 'Queued'},
+      {id: :working, name: 'Working'},
+      {id: :successful, name: 'Successful'},
+      {id: :failed, name: 'Failed'},
+      {id: :timed_out, name: 'Timed out'},
+      {id: :cancelled, name: 'Cancelled'},
+  ]
+
+  enumerize :status, in: AVAILABLE_ITEM_STATUS, predicates: true, default: :new
+
+  SYSTEM_JOB_ID = 'system'
+
+  #
+  # State transition map
+  #                             --> :successful
+  #                             |
+  # :new → :queued → :working ----> :failed
+  #   |       |          |      |
+  #   |       |          |      --> :timed_out
+  #   |       |          |
+  #   --------------------------> :cancelled
+  #
+  # IFF #retry implemented (not currently the case), then also:
+  #
+  #   :failed ---------> :queued
+  #                |
+  #   :timed_out ---
+
+  def status=(new_status)
+    old_status = self.status
+
+    # don't let enumerize set the default value when selecting nil from the database
+    new_status = nil if !new_record? && new_status == :new.to_s && old_status == nil
+
+    super(new_status)
+
+    update_status_timestamps(self.status, old_status)
+
+  end
+
+  def analysis_job_id
+    super_id = super()
+    return SYSTEM_JOB_ID if !new_record? && id.nil?
+
+    super_id
+  end
+
+  def self.filter_settings(is_system = false)
+
+    fields = [
+        :id, :analysis_job_id, :audio_recording_id,
+        :created_at, :queued_at, :work_started_at, :completed_at,
+        :queue_id,
+        :status
+    ]
+
+    settings = {
+        valid_fields: fields,
+        render_fields: fields,
+        text_fields: [:queue_id],
+        controller: :analysis_jobs_items,
+        action: :filter,
+        defaults: {
+            order_by: :audio_recording_id,
+            direction: :asc
+        },
+        valid_associations: [
```

Comment:
TODO: consider adding more nested relations
NB: available set to false for many-many tables